dbms/src/Common/OptimizedRegularExpression.inl.h

```diff
@@ -604,48 +609,55 @@ std::optional<StringRef> OptimizedRegularExpressionImpl<thread_safe>::substrImpl
 }
 
 template <bool thread_safe>
-void OptimizedRegularExpressionImpl<thread_safe>::replaceAllImpl(const char * subject, size_t subject_size, DB::ColumnString::Chars_t & res_data, DB::ColumnString::Offset & res_offset, const StringRef & repl, Int64 byte_pos)
+void OptimizedRegularExpressionImpl<thread_safe>::replaceAllImpl(const char * subject, size_t subject_size, DB::ColumnString::Chars_t & res_data, DB::ColumnString::Offset & res_offset, Int64 byte_pos, const Instructions & instructions)
 {
     size_t byte_offset = byte_pos - 1; // This is a offset for bytes, not utf8
     StringPieceType expr_sp(subject + byte_offset, subject_size - byte_offset);
     StringPieceType matched_str;
-    size_t prior_offset = 0;
+    size_t start_pos = 0;
+    size_t expr_len = expr_sp.size();
+    StringPieceType matches[MAX_CAPTURES];
+
+    // Copy characters that before position
+    res_data.resize(res_data.size() + byte_offset);
+    memcpy(&res_data[res_offset], subject, byte_offset);
+    res_offset += byte_offset;
 
     while (true)
     {
-        bool success = RegexType::FindAndConsume(&expr_sp, *re2, &matched_str);
+        bool success = re2->Match(expr_sp, start_pos, expr_len, re2_st::RE2::Anchor::UNANCHORED, matches, capture_num + 1);
```

Comment:
Since `matches` will be reused, I'm not sure if it need to be reset before `re2->Match`?

Reply:
> Since `matches` will be reused, I'm not sure if it need to be reset before `re2->Match`?

I think it's needless as the implementation in ch doesn't reset the `matches`